### PR TITLE
fix: modify bindata script to use curl

### PR DESCRIPTION
**Pull request by varshaprasad96**
**Description of the change:**
Use curl instead of wget
Closes: #4152 

**Motivation for the change:**
Mac does not have wget pre installed

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Review by joelanford — APPROVED** [review]
/lgtm

**Review by camilamacedo86 — APPROVED** [review]
/lgtm

File: Makefile [hunk 1]
```diff
@@ -41,7 +41,7 @@ generate: build # Generate CLI docs and samples
 	go run ./hack/generate/samples/generate_testdata.go
 
 .PHONY: bindata
-OLM_VERSIONS = 0.16.1 0.15.1
+OLM_VERSIONS = 0.17.0 0.16.1 0.15.1
```
Comment (varshaprasad96) [inline Makefile:44]: Well, we need not add modifications in the script if we specify the right tag here. ie: `OLM_VERSIONS = v0.17.0 0.16.1 0.15.1`

File: hack/generate/olm_bindata.sh [hunk 2]
```diff
@@ -1,9 +1,20 @@
 #!/usr/bin/env bash
 
+base_version=0.16.1
```
Comment (varshaprasad96) [inline hack/generate/olm_bindata.sh:3]: This is assuming, any release after 0.16.1 will have `v` appended to the tag.

File: Makefile [hunk 3]
```diff
@@ -41,7 +41,7 @@ generate: build # Generate CLI docs and samples
 	go run ./hack/generate/samples/generate_testdata.go
```
Comment (camilamacedo86) [inline Makefile:42]: Missing changelog

```
# entries is a list of entries to include in
# release notes and/or the migration guide
entries:
  - description: >
      Add OLM bindata for the 0.17.0 release version.
    # kind is one of:
    # - addition
    # - change
    # - deprecation
    # - removal
    # - bugfix
    kind: "addition"
    # Is this a breaking change?
    breaking: false

```

File: internal/olm/installer/client.go [hunk 4]
```diff
@@ -196,13 +196,13 @@ func (c Client) getResources(ctx context.Context, version string) ([]unstructure
...
+		crdResources, err = c.getCRDs(ctx, resolvedVersion)
 		if err != nil {
 			return nil, fmt.Errorf("failed to fetch CRDs: %v", err)
 		}
 
-		olmResources, err = c.getOLM(ctx, version)
+		olmResources, err = c.getOLM(ctx, resolvedVersion)
 		if err != nil {
 			return nil, fmt.Errorf("failed to fetch resources: %v", err)
 		}
```
Comment (camilamacedo86) [inline internal/olm/installer/client.go:208]: Missing update all e2e tests to use 0.17.0 now. See the changes made in the PR: https://github.com/operator-framework/operator-sdk/pull/4056/files

**Pull request merged**